app/javascript/packs/src/settings/components/team/components/TeamsMembers.jsx

```diff
@@ -0,0 +1,212 @@
+
+import RemoveUserModal from "./RemoveUserModal";
+import DataTable from "../../../../../shared/data_table";
+import { UPDATE_USER_TEAM_ROLE_PATH } from "../../../../../app/routes";
+
+const initalUserToRemove = {
+  userName: "",
+  team_user_id: 0,
+  teamName: "",
+  team_id: 0
+};
+class TeamsMembers extends Component {
+  constructor(params) {
+    super(params);
+    this.state = { showModal: false, userToRemove: initalUserToRemove };
+    this.memberAction = this.memberAction.bind(this);
+    this.hideModal = this.hideModal.bind(this);
+  }
+
+  currentRole(memberRole, role) {
+    return memberRole === role ? <Glyphicon glyph="ok" /> : "  ";
+  }
+
+  updateRole(userTeamId, role) {
+    axios
+      .put(UPDATE_USER_TEAM_ROLE_PATH, {
+        team: this.props.team.id,
+        user_team: userTeamId,
+        role
+      })
+      .then(response => {
+        this.props.updateUsersCallback(response.data.team_users);
+      })
+      .catch(error => console.log(error));
+  }
+
+  hideModal() {
+    this.setState({ showModal: false, userToRemove: initalUserToRemove });
+  }
+
+  userToRemove(userToRemove) {
+    this.setState({ showModal: true, userToRemove });
+  }
+
+  memberAction(data, row) {
+    return (
+      <DropdownButton
+        bsStyle="default"
+        disabled={data.disable}
+        title={
+          <span>
+            <Glyphicon glyph="cog" />
+          </span>
+        }
+        id="actions-dropdown"
+      >
+        <MenuItem header>
+          <FormattedMessage id="settings_page.single_team.actions.user_role" />
+        </MenuItem>
+        <MenuItem
+          onSelect={() => {
+            // 0 => Guest
+            this.updateRole(data.team_user_id, 0);
+          }}
+        >
+          {this.currentRole(data.current_role, "Guest")}
+          <FormattedMessage id="settings_page.single_team.actions.guest" />
+        </MenuItem>
+        <MenuItem
+          onSelect={() => {
+            // 1 => Normal user
+            this.updateRole(data.team_user_id, 1);
+          }}
+        >
+          {this.currentRole(data.current_role, "Normal user")}
+          <FormattedMessage id="settings_page.single_team.actions.normal_user" />
+        </MenuItem>
+        <MenuItem
+          onSelect={() => {
+            // 2 => Administrator
+            this.updateRole(data.team_user_id, 2);
+          }}
+        >
+          {this.currentRole(data.current_role, "Administrator")}
+          <FormattedMessage id="settings_page.single_team.actions.administrator" />
+        </MenuItem>
+        <MenuItem divider />
+        <MenuItem
+          onClick={() => {
+            this.userToRemove({
+              userName: row.name,
+              team_user_id: data.team_user_id,
+              teamName: this.props.team.name,
+              team_id: this.props.team.id
+            });
+          }}
+        >
+          <FormattedMessage id="settings_page.single_team.actions.remove_user" />
```

Comment:
settings_page.single_team.actions.remove_user // this probably violates low of Demeter